### ISSUE TITLE
TypeError: new.target does not define a custom element

### ISSUE THREAD

**Issue by kitaliev** [toeverything/blocksuite#8393]
**Description:**
I'm encountering an error when trying to run the `EdgelessEditor` component using version 0.17.10.

### **Safari:**
![image](https://github.com/user-attachments/assets/a61ce512-0ee6-48ac-b4b5-25a397683917)

### **Firefox and Chrome:**
![image](https://github.com/user-attachments/assets/5f9d258a-4a63-4abb-820f-366366e228c0)

**Code:**
```js
import { EdgelessEditor } from '@blocksuite/presets'
import { DocCollection, Schema } from '@blocksuite/store'
import { AffineSchemas } from '@blocksuite/blocks'
import '@blocksuite/presets/themes/affine.css'

const schema = new Schema().register(AffineSchemas)
const collection = new DocCollection({ schema })
collection.meta.initialize()

const doc = collection.createDoc({ id: 'page1' })
doc.load(() => {
  const pageBlockId = doc.addBlock('affine:page', {})
  doc.addBlock('affine:surface', {}, pageBlockId)
  const noteId = doc.addBlock('affine:note', {}, pageBlockId)
  doc.addBlock('affine:paragraph', {}, noteId)
})

const editor = new EdgelessEditor()
editor.doc = doc
document.body.appendChild(editor)
```

After some research, I found [this answer](https://stackoverflow.com/a/61883392), which suggests that `const editor = new EdgelessEditor()` might be the issue. So, I modified `main.js` as follows:

```js
import { EdgelessEditor } from '@blocksuite/presets'
import { DocCollection, Schema } from '@blocksuite/store'
import { AffineSchemas } from '@blocksuite/blocks'
import '@blocksuite/presets/themes/affine.css'

const schema = new Schema().register(AffineSchemas)
const collection = new DocCollection({ schema })
collection.meta.initialize()

const doc = collection.createDoc({ id: 'page1' })
doc.load(() => {
  // const pageBlockId = doc.addBlock('affine:page', {})
  // doc.addBlock('affine:surface', {}, pageBlockId)
  // const noteId = doc.addBlock('affine:note', {}, pageBlockId)
  // doc.addBlock('affine:paragraph', {}, noteId)
})

// const editor = new EdgelessEditor()
customElements.define('edgeless-editor', EdgelessEditor)
const editor = document.createElement('edgeless-editor')
editor.doc = doc
document.body.appendChild(editor)
```

However, I continued to get the same error until I commented out everything inside the `doc.load` callback.

Could you provide assistance with this issue?

I'm using
node 20.17
yarn 1.22
@blocksuite/blocks 0.17.10
@blocksuite/presets 0.17.10
@blocksuite/store 0.17.10
vite ^5.4.6
yjs ^13.6.19

**Comment by fourdim** [comment]
We have recently experienced a large change on how to define the custom element.
Previously, we used Lit's builtin decorator but it turns out to be not good for tree-shaking.
So we separate the `define the custom element` to a different function, which is `side effects` in functional programming.
In this case,  I would recommend you to run the side effects manually:

```ts
import { effects as blocksEffects } from '@blocksuite/blocks/effects';
import { effects as presetsEffects } from '@blocksuite/presets/effects';

blocksEffects();
presetsEffects();
```

Just like what we done in the playground:

https://github.com/toeverything/blocksuite/blob/723c326f91d09a2a66cebcb952d20f75db856571/packages/playground/apps/starter/main.ts#L32C1-L33C18
